### ISSUE TITLE
6.2.1 causes capitalization inconsistency 

### ISSUE THREAD

**Issue by alitasdln** [OWASP/ASVS#1875]
Other attacks in the document such as "brute force", "rainbow table", "phising" are all in lowercase but in [this requirement](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x14-V6-Cryptography.md#v62-algorithms) "Padding Oracle" is all in uppercase which may cause confusion for people who never heard about it. Especially considering Oracle is a well-established company in technology field. 

My proposal is to write it in all lowercase as other attack methods.

**Comment by elarlang** [comment]
We can change this requirement, but I think we need to set some rules in place and follow proper grammar in the document.

Does anyone have some good grammar rules to point out?

ping @tghosth @jmanico @Sjord

**Comment by jmanico** [comment]
Yes. Use a tool like Grammarly to check your grammar.

**Comment by elarlang** [comment]
> Yes. Use a tool like Grammarly to check your grammar.

not helpful in this case, as it does not know which words are names for something and which are not.

**Comment by Sjord** [comment]
> Padding Oracle

I agree that this should be lower case, and also the following should be lower case:

- libraries do not contain Easter eggs
- This should prevent Web Cache Deception attacks.
- Verify that the application is protected against ... Formula Injection
- to prevent ... Request Smuggling
- Verify that proper certification revocation, such as ... Stapling

I think in these cases uppercase is acceptable:
- JavaScript, LaTeX, Markdown, GraphQL, WebSocket
- CSV, HTTP, SMTP, API
- Content-Type, Set-Cookie

When an abbreviation is written in full it is often capitalized. I think these also should not be capitalized:
- Server-side Request Forgery (SSRF)
- Insecure Direct Object Reference (IDOR)

Currently all requirements start with "Verify that". I would rather omit that.

The first sentence of a requirement is often pretty clear and grammatically correct. The second sentence is often problematic; written for a different audience than the first, or gives a recommendation instead of a requirement:

- Registration and forgot password functionality should also have this protection.
- Strongly consider logging only in UTC if systems are global to assist with post-incident forensic analysis.
- Permissions should still be allocated using roles.

The requirements table should not have markdown formatting, because this doesn't work if it's exported to CSV or another format. Currently the links to proactive controls are the biggest problems here, and I suggest removing them.

**Comment by Sjord** [comment]
I think capitalization is partially subjective or a matter of taste, as opposed to a strict grammar rule that is right or wrong. There are several style guides with an opinion about capitals, and most instruct to use capitals sparingly:

- [Capitalization - Microsoft Style Guide | Microsoft Learn](https://learn.microsoft.com/en-us/style-guide/capitalization)
- [Capitalization  |  Google developer documentation style guide](https://developers.google.com/style/capitalization)
- [IEEE style guide](https://journals.ieeeauthorcenter.ieee.org/wp-content/uploads/sites/7/IEEE-Editorial-Style-Manual-for-Authors.pdf) p21
- [Wikipedia:Manual of Style/Capital letters - Wikipedia](https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style/Capital_letters)

There are also some existing OWASP style guides:

* [Style Guide - OWASP Mobile Application Security](https://mas.owasp.org/contributing/5_Style_Guide/)
* [wstg/style\_guide.md at master · OWASP/wstg](https://github.com/OWASP/wstg/blob/master/style_guide.md)

I think it would make sense for the ASVS or perhaps the whole of OWASP to pick an existing style guide as a base, and then add some clarifications to it. No need to reinvent the wheel.

**Comment by elarlang** [comment]
Yes, whatever are those rules, they just need to be applied through the document.

**Comment by tghosth** [comment]
Honestly I am happy for specific suggestions but I don't have strong opinions over this

**Comment by elarlang** [comment]
I would like to have 2 outcomes from this:
* The entire document fine-tuned
* All change decisions documented as the ruleset for the future

@Sjord - is this something you would like to do?

**Comment by elarlang** [comment]
Timeline-wise, maybe it makes sense to do this when we finalize the document. At the moment there is no point in putting effort into the chapter texts, only to the requirements.

**Comment by EnigmaRosa** [comment]
I'll add that Bishop Fox publishes a [cybersecurity style guide](https://images.bishopfox.com/prod-1437/Documents/Guides/Bishop-Fox-Cybersecurity-Style-Guide-V2.pdf) that might be of assistance. I'm open to compiling a style guide or doing grammar and style checks when ready.

**Comment by tghosth** [comment]
Good pointer @EnigmaRosa :)

This might be interesting when we are closer to a draft :)

**Comment by jmanico** [comment]
I agree that we should do this in one pass after we are close to a final draft.

**Comment by oleksiidov** [comment]
Colleagues, let me add some examples from discussion in #2892 for later review, when approach to style guide will be finalized: 
- proof key for code exchange (PKCE) in [10.1.2](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x19-V10-OAuth2-and-OIDC.md#v101-generic-oauth-and-oidc-security)
- Single Page Application (SPA) in [10.2](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x19-V10-OAuth2-and-OIDC.md#v102-oauth-client)
- pushed authorization request (PAR) in [10.4.12](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x19-V10-OAuth2-and-OIDC.md#v104-oauth-authorization-server)
- JWT-secured Authorization Request (JAR) in [10.4.12](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x19-V10-OAuth2-and-OIDC.md#v104-oauth-authorization-server)
- cross-site request forgery (CSRF) in [3.3.2](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x12-V3-Web-Frontend-Security.md#v33-cookie-setup) 
- local or remote file inclusion (LFI, RFI), and server-side request forgery (SSRF) in [5.3.2](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x14-V5-File-Handling.md#v53-file-storage)
-  time-based, one-time passwords (TOTPs) in [6.5.1](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x15-V6-Authentication.md#v65-general-multi-factor-authentication-requirements), [6.5.5](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x15-V6-Authentication.md#v65-general-multi-factor-authentication-requirements)
- cryptographically secure pseudo-random number generator (CSPRNG) in [7.2.4](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x16-V7-Session-Management.md#v72-fundamental-session-management-security), [11.5.1](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x20-V11-Cryptography.md#v115-random-values)
- insecure direct object reference (IDOR) and broken object level authorization (BOLA) in [8.2.2](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x17-V8-Authorization.md#v82-general-authorization-design)
- broken object property level authorization (BOPLA) in [8.2.3](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x17-V8-Authorization.md#v82-general-authorization-design)
- proof key for code exchange (PKCE) in [10.1.2](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x19-V10-OAuth2-and-OIDC.md#v101-generic-oauth-and-oidc-security), [10.2.1](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x19-V10-OAuth2-and-OIDC.md#v102-oauth-client), [10.4.6](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x19-V10-OAuth2-and-OIDC.md#v104-oauth-authorization-server) 
- cross-site request forgery (CSRF) in [10.2.1](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x19-V10-OAuth2-and-OIDC.md#v102-oauth-client)
- pushed authorization requests (PAR) in [10.4.12](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x19-V10-OAuth2-and-OIDC.md#v104-oauth-authorization-server), [10.4.13](https://github.com/OWASP/ASVS/blob/master/5.0/en/0x19-V10-OAuth2-and-OIDC.md#v104-oauth-authorization-server)

**Comment by jmanico** [comment]
I think all of these words should be capitalized. That's the standard in most (but not all) docs that I see. (And nice work pointing all of this out)

**Comment by oleksiidov** [comment]
Potentially, as we run out of time we could do the following:
At least for shortlisted terms above, agree on approach case-by-case. And at least make these changes. 
And leave the rest for the future. It may be kind of inconsistent, but it could improve the final document a little...

**Comment by tghosth** [comment]
For context, we need to release next week and I'm not available most days next week :)

I think first priority should be to uncover any inconsistencies where we capitalise the same phrase differently in different places. If there is none of that then you could see if there are any capitalisations which you strongly think are wrong and we can see how it goes.

**Comment by tghosth** [comment]
But @oleksiidov I tagged you in a different issue which is more urgent that honestly I prefer you worked on :)
